source/Phrase.cpp

```diff
@@ -151,6 +153,18 @@ Phrase::Choice::Choice(const DataNode &node, bool isPhraseName)
 
 
 
+// Pick a random choice from this part given their weights.
+const Phrase::Choice &Phrase::Part::Get() const
+{
+	unsigned index = 0;
+	for(int choice = Random::Int(total); choice >= choices[index].weight; ++index)
+		choice -= choices[index].weight;
+	
+	return choices[index];
+}
```

Comment:
It is something I’d thought of, so I’m certainly not opposed to the idea.

Reply:
It would make some level of sense to do so here, as you're taking code used in one place and using it in another. Why defer testing that code until after it's in more widespread use? (Deferring tests until later is a great way to never write tests at all.)

On the other hand, refactoring the fleet variant-picking code to use it is something that could easily be deferred to a follow-up PR.

Reply:
I imagine this will be merged before #4704, but so long as that PR is something we want merged, I'll probably just handle Fleet (and Variant) using WeightedList in that PR, since I'd need to update that PR to use it regardless.